proxy/plugin/cache/base.py

```diff
@@ -27,34 +32,109 @@ class BaseCacheResponsesPlugin(HttpProxyBasePlugin):
     Different storage backends can be used per request if required.
     """
 
+    class EnabledDescriptor:
+        def __init__(self, enabled: bool = False) -> None:
+            self.enabled = multiprocessing.Event()
+            if enabled:
+                self.enabled.set()
+
+        def __get__(self, obj: Optional[object], owner: type) -> Any:
+            if obj is None:
+                return self.enabled
+            return None
+
+    # Dynamically enable / disable cache
+    enabled = EnabledDescriptor(True)
```

Comment:
Also must note that dynamic `enabled` flag doesn't work on `MacOS` `Python 3.8`.  https://github.com/abhinavsingh/proxy.py/pull/425/checks?check_run_id=1197597150

I am also able to reproduce it locally.  Somehow the event changes are not propagated within plugin method checks.